components/switch/switch.go

```diff
@@ -51,16 +51,27 @@ type Switch interface {
 	GetNumberOfPositions(ctx context.Context, extra map[string]interface{}) (uint32, []string, error)
 }
 
-// FromRobot is a helper for getting the named Switch from the given Robot.
+// Deprecated: FromRobot is a helper for getting the named Switch from the given Robot.
+// Use FromProvider instead.
+//
+//nolint:revive // ignore exported comment check
 func FromRobot(r robot.Robot, name string) (Switch, error) {
 	return robot.ResourceFromRobot[Switch](r, Named(name))
 }
 
-// FromDependencies is a helper for getting the named button component from a collection of dependencies.
+// Deprecated: FromDependencies is a helper for getting the named button component from a collection of dependencies.
```

Comment:
(minor, flyby request) totally unrelated to your PR, but this looks like a bad historical copy/paste. Admittedly this is now deprecated but probably good to still change this from `button` to `switch`.